test/tests/functional/pbs_checkpoint.py

```diff
@@ -0,0 +1,83 @@
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.functional import *
+from ptl.utils.pbs_crayutils import CrayUtils
+
+
+class TestCheckpoint(TestFunctional):
+    """
+    This test suite targets Checkpoint functionality.
+    """
+
+    def test_checkpoint_abort_with_preemept(self):
+        abort_script = """#!/bin/bash
+kill $1
+exit 0
+"""
+        abort_file = self.du.create_temp_file(body=abort_script)
+        self.du.chmod(path=abort_file, mode=0755)
+        c = {'$action': 'checkpoint_abort 30 !' + abort_file + ' %sid'}
+        self.mom.add_config(c)
+        self.scheduler.set_sched_config({'preempt_order': 'C'})
+        a = {'queue_type': 'e',
+             'started': 'True',
+             'enabled': 'True',
+             'Priority': 200}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "expressq")
+        self.platform = self.du.get_platform()
+        if self.platform != 'cray' and self.platform != 'craysim':
+            node_attrs = self.server.status(NODE, id=self.server.shortname)[0]
+            ncpus = node_attrs['resources_available.ncpus']
+            attrs = {'Resource_List.select': '1:ncpus=' + ncpus}
+        else:
+            nv = CrayUtils().num_compute_vnodes()
+            attrs = {ATTR_l + '.select': '%d:ncpus=1' % nv,
+                     ATTR_l + '.place': 'scatter'}
+
+        j1 = Job(TEST_USER, attrs)
+        j1.set_sleep_time(1000)
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        attrs['queue'] = 'expressq'
+        j2 = Job(TEST_USER, attrs)
+        j2.set_sleep_time(1)
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid1)
+        self.server.delete(id=jid1)
```

Comment:
@vccardenas I don't agree with this statement.  @prakashcv13 has already written a test that reproduces the bug.  I see no reason for him to write a test for every way that the bug can be reproduced...

Reply:
The fix should be addressing all observed failure modes.  In order to show that, there need to be PTL tests for each of those.

Reply:
I agree with @lisa-altair.  There are three methods of triggering checkpointing.  All three send a checkpoint command to the mom.  Everything after that is the same.  @prakashcv13 changed code after the mom responds to the checkpoint command she was given.  He wrote a test which used one of the three methods of triggering checkpointing, since his fix has nothing to do with how one triggers checkpointing.  It has to do with how the job is checkpointed.

Reply:
The reasons to have the PTL tests for the bug fix include ensuring that the assumptions behind the fix are correct and that all affected scenarios beneft from the fix.  The fix may be independent of the triggering method, but the consequences of the fix could be different (or not).  The way to show which it is with a test.

Reply:
@iestockdale while i agree with you that we must have tests for the 3 different methods of checkpointing, i agree with the others that for this particular fix we do not need the contributor to cover those tests. What he has used to test well covers his area of change. I feel we should raise a separate bug to add test coverage in the other triggering methods, but that is beyond the scope of this fix alone.

Reply:
thank you everyone!